Mage.Sets/src/mage/cards/m/MegatronDestructiveForce.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    private MegatronDestructiveForceEffect(final MegatronDestructiveForceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MegatronDestructiveForceEffect copy() {
+        return new MegatronDestructiveForceEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        TargetPermanent target = new TargetPermanent(
+            0, 1, StaticFilters.FILTER_CONTROLLED_ANOTHER_ARTIFACT_SHORT_TEXT, true
+        );
+        player.choose(outcome, target, source, game);
+        Permanent permanent = game.getPermanent(target.getFirstTarget());
+        if (permanent == null) {
+            return false;
+        }
+
+        int manaValue = Math.max(permanent.getManaValue(), 0);
+        if (!permanent.sacrifice(source, game)) {
+            return false;
+        }
+
+        ReflexiveTriggeredAbility ability = new ReflexiveTriggeredAbility(
+            new MegatronDestructiveForceReflexiveEffect(manaValue), false
+        );
+        ability.addHint(new StaticHint("Sacrificed artifact mana value: " + manaValue));
+        ability.addTarget(new TargetCreaturePermanent());
+        game.fireReflexiveTriggeredAbility(ability, source);
+        return true;
+    }
+}
+
+class MegatronDestructiveForceReflexiveEffect extends OneShotEffect {
+
+    private final int value;
+
+    MegatronDestructiveForceReflexiveEffect(int value) {
+        super(Outcome.Damage);
+        staticText = "{this} deals damage equal to the sacrificed artifact's mana value to target " +
+            "creature. If excess damage would be dealt to that creature this way, instead that damage " +
+            "is dealt to that creature's controller and you convert {this}.";
+        this.value = value;
+    }
+
+    private MegatronDestructiveForceReflexiveEffect(final MegatronDestructiveForceReflexiveEffect effect) {
+        super(effect);
+        this.value = effect.value;
+    }
+
+    @Override
+    public MegatronDestructiveForceReflexiveEffect copy() {
+        return new MegatronDestructiveForceReflexiveEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent sourcePermanent = source.getSourcePermanentOrLKI(game);
+        if (sourcePermanent == null) {
+            return false;
+        }
+
+        if (value < 1) {
+            return false;
+        }
+
+        Permanent permanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+        if (permanent == null) {
+            // target should be a creature, thus a permanent.
+            // In case it is not, is it a player?
```

Comment:
Why are we trying to damage a player? I don't think this should happen

Reply:
I was scared of redirection of damage to another object. But I was confused for no reason.